st3/sublime_lib/resource_path.py

```diff
@@ -217,23 +246,18 @@ def with_suffix(self, suffix):
     def file_path(self):
         """
         Return a :class:`Path` object representing a filesystem path
-        inside the `Packages` or `Cache` directory.
+        inside one of Sublime's data directories.
 
         Even if there is a resource at this path,
         there may not be a file at that filesystem path.
         The resource could be in a default package or an installed package.
 
-        :raise ValueError: if the path's root is not
-        ``'Packages'`` or ``'Cache'``.
+        :raise ValueError: if the path's root is not used by Sublime.
         """
-        if self.root == 'Packages':
-            base = sublime.packages_path()
-        elif self.root == 'Cache':
-            base = sublime.cache_path()
-        else:
-            raise ValueError("%r is not a packages or cache path" % (self,))
-
-        return Path(base).joinpath(*self.parts[1:])
+        try:
+            return Path(RESOURCE_ROOTS[self.root]).joinpath(*self.parts[1:])
+        except KeyError:
+            raise ValueError("Can't find a filesystem path for {!r}.".format(self.root))
```

Comment:
```suggestion
            raise ValueError("Can't find a filesystem path for {!r}.".format(self.root)) from None
```

(might exceed maximum line lenght?)